spec/services/submit_application_spec.rb

```diff
@@ -2,19 +2,50 @@
 
 RSpec.describe SubmitApplication do
   describe 'Submit an application' do
-    it 'updates the application to Submitted and sets application_form.submitted_at' do
-      application_form = create(:application_form)
+    let(:application_form) { create(:application_form) }
+
+    before do
       create(:application_choice, application_form: application_form, status: 'unsubmitted')
       create(:application_choice, application_form: application_form, status: 'unsubmitted')
+    end
+
+    it 'updates the application to Submitted' do
+      SubmitApplication.new(application_form).call
+      expect(application_form.application_choices[0]).to be_awaiting_references
+      expect(application_form.application_choices[1]).to be_awaiting_references
+      expect(application_form.support_reference).not_to be_empty
+    end
+
+    context 'sending emails' do
+      let(:mailer) { double }
+
+      before { allow(mailer).to receive(:deliver_now) }
+
+      it 'sends an application submitted email' do
+        allow(CandidateMailer).to receive(:submit_application_email).and_return(mailer)
+
+        SubmitApplication.new(application_form).call
+        expect(CandidateMailer).to have_received(:submit_application_email).with(application_form)
+        expect(mailer).to have_received(:deliver_now)
+      end
+
+      it 'sends a reference request email to each referee' do
```

Comment:
This isn't, though. I wonder if it's better to have it in a referee-focussed system spec rather than here, anyway? (for example, If we change to async email delivery, which is likely, this spec will need to change)